src/registriesView.ts

```diff
@@ -97,6 +103,45 @@ export class ComponentTypesView implements TreeDataProvider<ComponentType> {
         return this.registries;
     }
 
+    public getCompDescriptions(): Set<ComponentTypeDescription> {
+        return this.compDescriptions;
+    }
+
+    public getListOfRegistries(): Registry[] {
+        return this.registries;
+    }
+
+    public getAllComponents(): void {
+        let isError = false;
+        this.compDescriptions.clear();
+        void getInstance().getCompTypesJson().then(async (devFileComponentTypes: DevfileComponentType[]) => {
```

Comment:
There should be catch in the chain for getCompTypesJson() call. Catch callback should show error message with info that odo could not pull component types form registry.